dotCMS/src/main/java/com/dotcms/rest/config/DotRestApplication.java

```diff
@@ -87,7 +89,8 @@ public static synchronized void addClass(final Class<?> clazz) {
 			return;
 		}
 		if (Boolean.TRUE.equals(customClasses.computeIfAbsent(clazz,c -> true))) {
-			ContainerReloader.getInstance().reload();
+			final Optional<ContainerReloader> reloader = CDIUtils.getBean(ContainerReloader.class);
+            reloader.ifPresent(ContainerReloader::reload);
```

Comment:
Does this need to error out if `CDIUtils.getBean(ContainerReloader.class).isEmpty()` ?

Reply:
It should never be empty, right?

Reply:
true should never.. but in this case the issue came  from my "fix" it was missing the @Provider Annotation leading to a non active app listener, the Context was arriving null and the reload never took place